### DISCUSSION TITLE
Qt Creator and Stellarium-master

### DISCUSSION THREAD

**Discussion by astromout** [Stellarium/stellarium#1635] [Q&A] [answered]
Hi, 

I'm happy to be here...I tried scripts on stellarium. It works well. 

But now I wanted to get a kind of "virtual desktop" in such away as to have "the public viewer" on a screen (videoprojector) and the "chief viewer" on my screen.

So, I want to put my hand in stellarium's code...
I downloaded stellarium-master.
I work with Qt-creator. 
But, I have these errors :
CMake Error at CMakeLists.txt:18 (PROJECT):
  The CMAKE_C_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  To use the JOM generator with Visual C++, cmake must be run from a shell
  that can use the compiler cl from the command line.  This environment is
  unable to invoke the cl compiler.  To fix this problem, run cmake from the
  Visual Studio Command Prompt (vcvarsall.bat).

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.

CMake Error at CMakeLists.txt:18 (PROJECT):
  The CMAKE_CXX_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  To use the JOM generator with Visual C++, cmake must be run from a shell

...

I read lot's of thing on internet around this error, but I'm like a chicken in front of a knife.
Help!! 
Regards.
Franck.

**Comment by gzotti** [comment]
Have you read the BUILDING.md instructions?

For your original problem, you can use SPOUT output to your projector, or use the RemoteControl HTTP interface.

**Reply by astromout** [reply]
Hi, 
Thank you so much for helping...Thanks to you, I looked à BUILDING.md instructions and tried.
A priori, PATHS are ok (like in building.md), cmake (3.19.2), 
Some errors appeared even in command line. They came at the end of compilation 

Here are they:
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\indicom.h(45,1): warning C4005: '_USE_MATH_DEFINES' :
 redéfinition de macro (compilation du fichier source C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs
\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclient.vcxproj]
ligne de commande(1): message : voir la définition précédente de '_USE_MATH_DEFINES' (compilation du fichier source C:\
Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\buil
d\msvc\src\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(45,25): warning C5104: 'L#s' a été tr
ouvé dans la liste de remplacements de macros. Vouliez-vous dire 'L""#s' ? (compilation du fichier source C:\Stellarium
-Mai2021\stellarium-master\src\external\libindi\libs\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src
\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(81,39): error C2065: 'L' : identifica
teur non déclaré (compilation du fichier source C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\indic
om.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(81,39): warning C4047: 'fonction' : l
es niveaux d'indirection de 'const locale_char_t *' et de 'int' sont différents (compilation du fichier source C:\Stell
arium-Mai2021\stellarium-master\src\external\libindi\libs\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\build\msv
c\src\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(81,39): warning C4024: 'indi_setlocal
e' : types différents pour le paramètre formel et réel 2 (compilation du fichier source C:\Stellarium-Mai2021\stellariu
m-master\src\external\libindi\libs\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclie
nt.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(81,39): error C2143: erreur de syntax
e : absence de ')' avant 'chaîne' (compilation du fichier source C:\Stellarium-Mai2021\stellarium-master\src\external\l
ibindi\libs\indicom.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\locale_compat.h(81,55): error C2059: erreur de syntax
e : ')' (compilation du fichier source C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\indicom.c) [C:
\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\lilxml.c(508,1): warning C4267: 'initialisation' : co
nversion de 'size_t' en 'int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\extern
al\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\lilxml.c(784,75): warning C4267: '+=' : conversion de
 'size_t' en 'int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclie
nt.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\lilxml.c(797,49): warning C4267: '+=' : conversion de
 'size_t' en 'int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\indiclie
nt.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\lilxml.c(866,1): warning C4267: 'initialisation' : co
nversion de 'size_t' en 'int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\extern
al\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\lilxml.c(1255,1): warning C4267: 'initialisation' : c
onversion de 'size_t' en 'int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\exter
nal\indiclient.vcxproj]
C:\Stellarium-Mai2021\stellarium-master\src\external\libindi\libs\indicom.c(442,25): warning C4267: '=' : conversion de
 'size_t' en 'unsigned int', perte possible de données [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external
\indiclient.vcxproj]

**Reply by alex-w** [reply]
Please try disable Telescope Control plugin in CMakeLists.txt file

**Reply by astromout** [reply, marked as answer]
I will try this, but, this morning, we compile fine and stellarium.exe was created.
After we tried to build with Qt Created and...errors appeared.

**Reply by astromout** [reply]
Hi, 
I tried without Telescope Control plugin in CMakelists.txt and it compiles! Thank you so much.
In cmd, I just have some warnigs (in yellow). For example : 

C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\winbase.h(9531,5): warning C5105: l'expansion de macro produisant 'defined' a un comportement indéfini (compilation du fichier source C:\Stellarium
-Mai2021\stellarium-master\src\external\glues_stel\source\libtess\tess.c) [C:\Stellarium-Mai2021\stellarium-master\build\msvc\src\external\glues_stel.vcxproj]

- Are these warnings are important?

- When I launch stellarium.exe, my computer tells me SpoutLibrary.dll is missing...so I copy it in stellarium.exe folder and it works.

**Reply by astromout** [reply]
Hi, 

Stellarium runs with cmd and with Qt Creator ❤️  
I'm so happy! Thank you for your help...

Is it hard to have the step to get Telescope Control working ?

**Reply by gzotti** [reply]
Not sure. For me it worked out of the box on 5 Windows and 2 Ubuntu systems. Obstacles should be described in the BUILDING.md.

**Reply by astromout** [reply]
OK, 
I have so much things to understand...Telescope Control will wait.

**Comment by astromout** [comment]
Parentheses : I play the RemoteControl HTTP interface. It is a fabulous plugin, exactly what I want...Of course, I will try to add things on it as soon as I'll be able to compile.
